### PR TITLE
MNT Skip tests when session-manager module is installed

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/recipe-cms/issues/47

### CODE REVIEW

File: tests/php/Security/MemberTest.php [hunk 1]
```diff
@@ -1198,18 +1199,21 @@ public function testRememberMeMultipleDevices()
...
-        // Logging out from any device when all login hashes should be removed
-        RememberLoginHash::config()->update('logout_across_devices', true);
-        Injector::inst()->get(IdentityStore::class)->logIn($m1, true);
-        $this->get('Security/logout', $this->session());
-        $this->assertEquals(
-            RememberLoginHash::get()->filter('MemberID', $m1->ID)->count(),
-            0
-        );
+        // If session-manager module is installed then logout_across_devices is modified so skip
+        if (!class_exists(LoginSession::class)) {
```
Comment (maxime-rainville) [inline tests/php/Security/MemberTest.php:1207]: Is the behaviour when session manager installed predictable? Maybe we want to add an opposing test here or in session manager.

File: tests/php/Security/RememberLoginHashTest.php [hunk 2]
```diff
@@ -49,6 +50,10 @@ public function clearScenarios()
      */
     public function testClear(bool $logoutAcrossDevices, string $deviceId, array $expected, array $unexpected)
     {
+        // If session-manager module is installed then logout_across_devices is modified so skip
+        if (class_exists(LoginSession::class)) {
```
Comment (maxime-rainville) [inline tests/php/Security/RememberLoginHashTest.php:54]: What's the expected behaviour if Session-Manager is installed?

File: tests/php/Security/RememberLoginHashTest.php [hunk 3]
```diff
@@ -75,6 +80,10 @@ public function testClear(bool $logoutAcrossDevices, string $deviceId, array $ex
 
     public function testGetSetLogoutAcrossDevices()
     {
+        // If session-manager module is installed then logout_across_devices is modified so skip
+        if (class_exists(LoginSession::class)) {
```
Comment (maxime-rainville) [inline tests/php/Security/RememberLoginHashTest.php:84]: What's the expected behaviour if Session-Manager is installed?

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
I'm only happy with skipping these tests altogether if they become non-sensical to run when session-manager is installed.

**Comment by emteknetnz** [comment]
@maxime-rainville these tests are non-nonsensical IMO when session-manager is installed, because of this line in the session-manager middleware

`RememberLoginHash::setLogoutAcrossDevices(false);`
https://github.com/silverstripe/silverstripe-session-manager/blob/1/src/Middleware/LoginSessionMiddleware.php#L39

The tests are explicitly modifying `logout_across_devices` and then running an assertion.  However session-manager will just override what the test set `logout_across_devices` to

File: tests/php/Security/MemberTest.php [hunk 4]
```diff
@@ -1198,18 +1199,21 @@ public function testRememberMeMultipleDevices()
             ]
         );
         $this->assertEquals(
+            1,
             RememberLoginHash::get()->filter(['MemberID'=>$m1->ID, 'DeviceID'=>$firstHash->DeviceID])->count(),
```
Comment (maxime-rainville) [inline tests/php/Security/MemberTest.php:1203]:
```suggestion
            RememberLoginHash::get()->filter(['MemberID'=>$m1->ID, 'DeviceID'=>$firstHash->DeviceID])->count()
```

**Review by maxime-rainville — APPROVED** [review]
Just need to fix that rogue comma

**Comment by maxime-rainville** [comment]
I fixed the comma and updated the build to run recipe-cms 4.9.x just to keep this moving.

**Pull request merged**